### ISSUE TITLE
Sportliveevents does not work on kodi 18

### ISSUE THREAD

**Issue by pitsi** [mbebe/blomqvist#780]
First of all, although I rarely watch sports, thank you for that new addon :) 

I installed it on my everyday kodi 18.9 installation (libreelec 9.2.6 x64) and I wanted to try it out. As it seems, nothing but the main menu is opening, because scrapping fails on every single entry as seen in the log below
```
2022-05-23 09:07:14.913 T:140474950051584   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ImportError'>
                                            Error Contents: No module named request
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 301, in <module>
                                                router(sys.argv[2][1:])
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 133, in router
                                                mod = import_mod(mode2)
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 91, in import_mod
                                                if   s == 'cric': from resources.lib import cricfree as mod
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/resources/lib/cricfree.py", line 15, in <module>
                                                from resources.lib.cmf3 import parseDOM
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/resources/lib/cmf3.py", line 20, in <module>
                                                import urllib.request, urllib.parse, urllib.error
                                            ImportError: No module named request
                                            -->End of Python script error report<--
2022-05-23 09:07:15.051 T:140474916480768   ERROR: GetDirectory - Error getting plugin://plugin.video.sportliveevents/?ilabel=%7b%27title%27%3a%20%27CricFree%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2fimages%2fcrfree.png&mode=listmenu%3acric&page=1&title=CricFree&url
2022-05-23 09:07:15.062 T:140476330502272   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.sportliveevents/?ilabel=%7b%27title%27%3a%20%27CricFree%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2fimages%2fcrfree.png&mode=listmenu%3acric&page=1&title=CricFree&url) failed
```
The addon works with no issues at all on kodi 19 (libreelec 10.0.2 x64, on the same system), so I assume it's some python command that fails on python2 but works on python3. If it helps, le 10 uses python 3.8.9 and le 9 uses python 2.7.16.
If it is not possible to fix it, please add this bit in adddon.xml to make it python3-only and now show up under kodi 18 (or older).
`<import addon="xbmc.python" version="3.0.0"/>`

Thank you in advance.

**Comment by mbebe** [comment]
Updated.
New version - 1.5 should be ok in K18.

**Comment by pitsi** [comment]
Sorry for the delay, but I just launched my kodi to verify it was fixed, so I am closing the issue. Thank you :)

**Comment by pitsi** [comment]
Sorry for bringing this back up, but there are a couple of issues I found. 

On both kodi 18 and 19, daddy live > schedule returns nothing and this shows up in the log
```
2022-06-12 19:11:19.634 T:139812550317824   ERROR: Previous line repeats 2 times.
2022-06-12 19:11:19.635 T:139812550317824   ERROR: GetDirectory - Error getting plugin://plugin.video.sportliveevents/?ilabel=%7bu%27plot%27%3a%20u%27Sunday%2012th%20June%202022%20%27%2c%20u%27code%27%3a%20None%2c%20u%27title%27%3a%20u%27Sunday%2012th%20June%202022%20%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2f..%2ficon.png&mode=listschedule%3adaddy&page=1&title=Sunday%2012th%20June%202022%20&url=Sunday%252012th%2520June%25202022%2520-%2520Schedule%2520Time%2520UK%2520GMT%252B1%7ccateg
2022-06-12 19:11:19.640 T:139812723001472   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.sportliveevents/?ilabel=%7bu%27plot%27%3a%20u%27Sunday%2012th%20June%202022%20%27%2c%20u%27code%27%3a%20None%2c%20u%27title%27%3a%20u%27Sunday%2012th%20June%202022%20%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2f..%2ficon.png&mode=listschedule%3adaddy&page=1&title=Sunday%2012th%20June%202022%20&url=Sunday%252012th%2520June%25202022%2520-%2520Schedule%2520Time%2520UK%2520GMT%252B1%7ccateg) failed
2022-06-12 19:11:21.295 T:139811146544896   ERROR: /storage/.kodi/addons/script.module.urllib3/lib/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              InsecureRequestWarning,
```
Moreover, stream3watch > schedule returns nothing and pops this in the log
```
2022-06-12 19:14:08.859 T:139812550317824   ERROR: Previous line repeats 2 times.
2022-06-12 19:14:08.859 T:139812550317824   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.IndexError'>
                                            Error Contents: list index out of range
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 319, in <module>
                                                router(sys.argv[2][1:])
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 161, in router
                                                links = mod.ListSchedule(exlink)
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/resources/lib/stream2watch.py", line 88, in ListSchedule
                                                html = parseDOM(html,'div', attrs={'class': "dropdown-content"})[0]##
                                            IndexError: list index out of range
                                            -->End of Python script error report<--
2022-06-12 19:14:08.996 T:139811160913664   ERROR: GetDirectory - Error getting plugin://plugin.video.sportliveevents/?ilabel=%7bu%27plot%27%3a%20%27Cricfree%20-%20schedule%27%2c%20u%27title%27%3a%20%27Schedule%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2fimages%2fschedule.png&mode=listschedule%3astream2watch&page=1&title=Schedule&url=dzien
2022-06-12 19:14:09.008 T:139812723001472   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.sportliveevents/?ilabel=%7bu%27plot%27%3a%20%27Cricfree%20-%20schedule%27%2c%20u%27title%27%3a%20%27Schedule%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2fimages%2fschedule.png&mode=listschedule%3astream2watch&page=1&title=Schedule&url=dzien) failed
```

And kubsports > schedule fails on kodi 18 only with this error in the log
```
2022-06-12 19:13:19.729 T:139812558710528   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ImportError'>
                                            Error Contents: No module named html
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 319, in <module>
                                                router(sys.argv[2][1:])
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 151, in router
                                                mod = import_mod(mode2)
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/main.py", line 113, in import_mod
                                                elif s == 'klubsports': from resources.lib import klubsports as mod
                                              File "/storage/.kodi/addons/plugin.video.sportliveevents/resources/lib/klubsports.py", line 13, in <module>
                                                import html
                                            ImportError: No module named html
                                            -->End of Python script error report<--
2022-06-12 19:13:19.874 T:139812550317824   ERROR: GetDirectory - Error getting plugin://plugin.video.sportliveevents/?ilabel=%7bu%27title%27%3a%20u%27KlubSports%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2f..%2ficon.png&mode=listmenu%3aklubsports&page=1&title=KlubSports&url
2022-06-12 19:13:19.890 T:139812723001472   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.sportliveevents/?ilabel=%7bu%27title%27%3a%20u%27KlubSports%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2f..%2ficon.png&mode=listmenu%3aklubsports&page=1&title=KlubSports&url) failed
```
Than you in advance.

**Comment by pitsi** [comment]
And fifa plus also fails to show the live now menu, in both 18 and 19, with this in the log
```
2022-06-18 11:19:07.414 T:139842113910528   ERROR: /storage/.kodi/addons/script.module.urllib3/lib/urllib3/connectionpool.py:1050: InsecureRequestWarning: Unverified HTTPS request is being made to host 'mcls-api.mycujoo.tv'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
                                              InsecureRequestWarning,
2022-06-18 11:19:14.176 T:139842113910528   ERROR: Previous line repeats 1 times.
2022-06-18 11:19:14.176 T:139842113910528   ERROR: GetDirectory - Error getting plugin://plugin.video.sportliveevents/?ilabel=%7bu%27plot%27%3a%20%27FIFA%20Plus%20-%20live%20now%27%2c%20u%27title%27%3a%20%27Live%20now%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2fimages%2fschedule.png&mode=listschedule%3afifaplus&page=1&title=Live%20now&url=cujo
2022-06-18 11:19:14.192 T:139842412480640   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.sportliveevents/?ilabel=%7bu%27plot%27%3a%20%27FIFA%20Plus%20-%20live%20now%27%2c%20u%27title%27%3a%20%27Live%20now%27%7d&image=%2fstorage%2f.kodi%2faddons%2fplugin.video.sportliveevents%2fresources%2fimages%2fschedule.png&mode=listschedule%3afifaplus&page=1&title=Live%20now&url=cujo) failed
```
It probably has to do with the recent update to urllib3...

**Comment by pitsi** [comment]
Moreover, I noticed that the dates under sporthd live > schedule are stuck to ~1 week ago (Saturday 11th to Monday 13th of June). Sadly, these are the dates that the site also has available for streams, so something is wrong with the site itself. I hope it resumes soon.

**Comment by pitsi** [comment]
Sadly, the recent update to 1.6 fixed none of the above issues, neither on kodi 18 nor on 19.

**Comment by pitsi** [comment]
I am delighted to say that all the above issues are fixed in v1.9! Thank you very much!

As for the issue on klubsports that happens only on kodi 18, it is because its python2 does not have the html library as mentioned in line 13 of `plugin.video.sportliveevents/resources/lib/klubsports.py`.
In order to make it python2 compatible, you must replace that line with these 2
```
from HTMLParser import HTMLParser
html = HTMLParser()

```
